src/services/expense.service.js

```
@@ -0,0 +1,63 @@
+const { v4: uuidv4 } = require('uuid');
+const { Op } = require('sequelize');
+const { Expense } = require('../models/models');
+
+async function getAll(userId, categories, dateFrom, dateTo) {
+  return Expense.findAll({
+    where: {
+      userId: userId,
+      categories: categories,
+      date: {
+        [Op.gte]: dateFrom,
+        [Op.lte]: dateTo,
+      },
+    },
+  });
+}
+
+async function getSingle(id) {
+  return Expense.findByPk(id);
+}
+
+async function create({ userId, title, amount, category, note }) {
+  const date = new Date();
+  const id = uuidv4();
+
+  return Expense.create({
+    id,
+    userId,
+    date,
+    title,
+    amount,
+    category,
+    note,
+  });
```

`create` sets and writes `date`, `title`, `amount`, `category`, and `note`. The current model lacks `date`/`title`/`amount`/`category`/`note` columns (it has `dateFrom`/`dateTo` and `categories`). This will cause `Expense.create(...)` to fail or silently ignore columns. Update the model to match these fields or change the service to create rows using the actual model columns.